{
  errno: undefined,
  code: 'ETIMEOUT',
  syscall: 'querySrv',
  hostname: '_mongodb._tcp.cluster0.n1j80p8.mongodb.net',
  level: 'error',
  message: '❌ Failed to Connect querySrv ETIMEOUT _mongodb._tcp.cluster0.n1j80p8.mongodb.net',
  stack: 'Error: querySrv ETIMEOUT _mongodb._tcp.cluster0.n1j80p8.mongodb.net\n' +
    '    at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/promises:251:17)',
  label: 'UMS',
  timestamp: '2023-06-10T10:05:13.126Z'
}
{
  errno: undefined,
  code: 'ETIMEOUT',
  syscall: 'querySrv',
  hostname: '_mongodb._tcp.cluster0.n1j80p8.mongodb.net',
  level: 'error',
  message: '❌ Failed to Connect querySrv ETIMEOUT _mongodb._tcp.cluster0.n1j80p8.mongodb.net',
  stack: 'Error: querySrv ETIMEOUT _mongodb._tcp.cluster0.n1j80p8.mongodb.net\n' +
    '    at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/promises:251:17)',
  label: 'UMS',
  timestamp: '2023-06-10T10:06:37.031Z'
}
{
  statusCode: 400,
  level: 'error',
  label: 'UMS',
  timestamp: '2023-06-10T10:29:48.199Z'
}
{
  statusCode: 409,
  level: 'error',
  label: 'UMS',
  timestamp: '2023-06-10T10:34:13.076Z'
}
{
  message: "❌ Failed to Connect Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/",
  reason: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    servers: Map(3) {
      'ac-kp5tuoh-shard-00-00.n1j80p8.mongodb.net:27017' => ServerDescription {
        address: 'ac-kp5tuoh-shard-00-00.n1j80p8.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        lastUpdateTime: 1050578,
        lastWriteDate: 0,
        error: MongoNetworkError: getaddrinfo ENOTFOUND ac-kp5tuoh-shard-00-00.n1j80p8.mongodb.net
            at connectionFailureError (C:\level2\ums\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:491:14)
            at TLSSocket.<anonymous> (C:\level2\ums\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:392:16)
            at Object.onceWrapper (node:events:628:26)
            at TLSSocket.emit (node:events:513:28)
            at emitErrorNT (node:internal/streams/destroy:151:8)
            at emitErrorCloseNT (node:internal/streams/destroy:116:3)
            at processTicksAndRejections (node:internal/process/task_queues:82:21) {
          cause: Error: getaddrinfo ENOTFOUND ac-kp5tuoh-shard-00-00.n1j80p8.mongodb.net
              at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26) {
            errno: -3008,
            code: 'ENOTFOUND',
            syscall: 'getaddrinfo',
            hostname: 'ac-kp5tuoh-shard-00-00.n1j80p8.mongodb.net'
          },
          [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        primary: null,
        me: null,
        '$clusterTime': null
      },
      'ac-kp5tuoh-shard-00-01.n1j80p8.mongodb.net:27017' => ServerDescription {
        address: 'ac-kp5tuoh-shard-00-01.n1j80p8.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        lastUpdateTime: 1050587,
        lastWriteDate: 0,
        error: MongoNetworkError: getaddrinfo ENOTFOUND ac-kp5tuoh-shard-00-01.n1j80p8.mongodb.net
            at connectionFailureError (C:\level2\ums\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:491:14)
            at TLSSocket.<anonymous> (C:\level2\ums\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:392:16)
            at Object.onceWrapper (node:events:628:26)
            at TLSSocket.emit (node:events:513:28)
            at emitErrorNT (node:internal/streams/destroy:151:8)
            at emitErrorCloseNT (node:internal/streams/destroy:116:3)
            at processTicksAndRejections (node:internal/process/task_queues:82:21) {
          cause: Error: getaddrinfo ENOTFOUND ac-kp5tuoh-shard-00-01.n1j80p8.mongodb.net
              at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26) {
            errno: -3008,
            code: 'ENOTFOUND',
            syscall: 'getaddrinfo',
            hostname: 'ac-kp5tuoh-shard-00-01.n1j80p8.mongodb.net'
          },
          [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        primary: null,
        me: null,
        '$clusterTime': null
      },
      'ac-kp5tuoh-shard-00-02.n1j80p8.mongodb.net:27017' => ServerDescription {
        address: 'ac-kp5tuoh-shard-00-02.n1j80p8.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        lastUpdateTime: 1050590,
        lastWriteDate: 0,
        error: MongoNetworkError: getaddrinfo ENOTFOUND ac-kp5tuoh-shard-00-02.n1j80p8.mongodb.net
            at connectionFailureError (C:\level2\ums\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:491:14)
            at TLSSocket.<anonymous> (C:\level2\ums\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:392:16)
            at Object.onceWrapper (node:events:628:26)
            at TLSSocket.emit (node:events:513:28)
            at emitErrorNT (node:internal/streams/destroy:151:8)
            at emitErrorCloseNT (node:internal/streams/destroy:116:3)
            at processTicksAndRejections (node:internal/process/task_queues:82:21) {
          cause: Error: getaddrinfo ENOTFOUND ac-kp5tuoh-shard-00-02.n1j80p8.mongodb.net
              at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26) {
            errno: -3008,
            code: 'ENOTFOUND',
            syscall: 'getaddrinfo',
            hostname: 'ac-kp5tuoh-shard-00-02.n1j80p8.mongodb.net'
          },
          [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        primary: null,
        me: null,
        '$clusterTime': null
      }
    },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: 'atlas-wdtraj-shard-0',
    maxElectionId: null,
    maxSetVersion: null,
    commonWireVersion: 0,
    logicalSessionTimeoutMinutes: null
  },
  code: undefined,
  level: 'error',
  stack: "MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/\n" +
    '    at _handleConnectionErrors (C:\\level2\\ums\\university-management-auth-service\\node_modules\\mongoose\\lib\\connection.js:792:11)\n' +
    '    at NativeConnection.openUri (C:\\level2\\ums\\university-management-auth-service\\node_modules\\mongoose\\lib\\connection.js:767:11)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:95:5)',
  label: 'UMS',
  timestamp: '2023-06-10T10:40:41.614Z'
}
{
  statusCode: 409,
  level: 'error',
  label: 'UMS',
  timestamp: '2023-06-10T10:45:26.493Z'
}
{
  errno: undefined,
  code: 'ETIMEOUT',
  syscall: 'queryTxt',
  hostname: 'cluster0.n1j80p8.mongodb.net',
  level: 'error',
  message: '❌ Failed to Connect queryTxt ETIMEOUT cluster0.n1j80p8.mongodb.net',
  stack: 'Error: queryTxt ETIMEOUT cluster0.n1j80p8.mongodb.net\n' +
    '    at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/promises:251:17)',
  label: 'UMS',
  timestamp: '2023-06-10T10:46:58.461Z'
}
{
  statusCode: 400,
  level: 'error',
  label: 'UMS',
  timestamp: '2023-06-10T10:51:51.820Z'
}
